transmart-oauth/grails-app/services/org/transmart/oauth/authentication/AuditLogService.groovy

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright © 2013-2016 The Hyve B.V.
+ *
+ * This file is part of Transmart.
+ *
+ * Transmart is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * Transmart.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.transmart.oauth.authentication
+
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import grails.core.GrailsApplication
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.oauth2.provider.OAuth2Authentication
+import org.transmartproject.core.users.User
+
+import javax.servlet.http.HttpServletRequest
+
+/**
+ * Logs to level TRACE in a specific `metric' format.
+ * Configuration of output done in (out-of-tree) Config.groovy.
+ */
+@CompileStatic
+@Slf4j
+class AuditLogService {
```

Comment:
This class now exists in transmart-oauth and transmartApp. Also this is not authentication related, so transmart-oauth is not the right place for this to live. Maybe the right place for this is transmart-server.